crates/xmtp-networking/src/grpc_api_helper.rs

```diff
@@ -35,23 +117,46 @@ impl Client {
             .parse()
             .map_err(|e| tonic::Status::new(tonic::Code::Internal, format!("{}", e)))?;
 
-        let mut client =
-            MessageApiClient::with_interceptor(self.channel.clone(), move |mut req: Request<()>| {
-                req.metadata_mut().insert("authorization", token.clone());
-                Ok(req)
-            });
-
-        return client
-            .publish(PublishRequest { envelopes })
-            .await
-            .map(|r| r.into_inner());
+        match &mut self.conn {
+            InnerApiConnection::Plain(c) => {
+                return MessageApiClient::with_interceptor(c, move |mut req: Request<()>| {
+                    req.metadata_mut().insert("authorization", token.clone());
+                    Ok(req)
+                })
+                .publish(PublishRequest { envelopes })
+                .await
+                .map(|r| r.into_inner());
+            }
+            InnerApiConnection::Tls(c) => {
+                return MessageApiClient::with_interceptor(c, move |mut req: Request<()>| {
+                    req.metadata_mut().insert("authorization", token.clone());
+                    Ok(req)
+                })
+                .publish(PublishRequest { envelopes })
+                .await
+                .map(|r| r.into_inner());
+            }
+        };
     }
 
     pub async fn subscribe(&mut self, topics: Vec<String>) -> Result<Subscription, tonic::Status> {
         let request = SubscribeRequest {
             content_topics: topics,
         };
-        let stream = self.client.subscribe(request).await.unwrap().into_inner();
+        let stream = match &self.client {
+            InnerApiClient::Plain(c) => c
+                .clone()
```

Comment:
I'm doing a lot more cloning than I would like, but I get compiler errors without it.